### PR TITLE
Add open62541

### PR DESCRIPTION
**Pull request by traversaro**
[open62541](https://open62541.org/) is one of the major open source implementation of the [OPC Unified Architecture](https://opcfoundation.org/about/opc-technologies/opc-ua/) protocol, a communication protocol widely used in industrial automation. 

One of the interesting characteristics of the OPC UA standard implemented by `open62541` is that the companies in various industrial domains, such as [Commercial Kitchen Equipment](https://opcfoundation.org/markets-collaboration/hki/) or [Industrial Robotics](https://opcfoundation.org/markets-collaboration/robotics/) define their own domain-specific standards (called ["Information Models"](https://opcfoundation.org/developer-tools/specifications-opc-ua-information-models)) on top of OPC UA standard. 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by traversaro** [comment]
The PR is now ready to review.

cc @conda-forge/help-c-cpp as it is a C library 
cc @wolfv as it is a robotics related library (even if not directly ROS-related)

File: recipes/open62541/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}              # [not win]
+    - {{ compiler('c') }}                # [not win]
+    - vs2017_win-64                      # [win64]
```
Comment (wolfv) [inline recipes/open62541/meta.yaml:21]: This shouldn't be necessary as cf is on vs 2017 by default now

**Review by wolfv — CHANGES_REQUESTED** [review]
Looks good!

File: recipes/open62541/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipes/open62541/meta.yaml]: Fixed in https://github.com/conda-forge/staged-recipes/pull/13231/commits/11fa452fc9da8680e5e03eebd1d7fc60214f4006 . By the way as this is a C-only library, also the dependency on a C++ compiler was not necessary.

Comment (traversaro) [reply recipes/open62541/meta.yaml]: Actually as CMake is not properly configured to look just for a C compiler via: 
~~~
project(... LANGUAGES C)
~~~
it uses the default setting (i.e. `C` and `CXX` as a enabled languages) so a C++ compiler is actually necessary.

**Pull request merged**